src/main/webapp/ts/matches.ts

```diff
@@ -1,16 +1,36 @@
-function loadMatches() {
+type authInfo = {
+    loginUrl: string;
+    logoutUrl: string;
+};
+
+async function loadMatches() {
+    const logStatus = await getLogStatus();
+
     fetch('/matches')
         .then(response => response.json())
         .then((matches) => {
+            if((logStatus.loginUrl === "") && (matches != null)) {
             const matchListElement = document.getElementById('match-history')!;
             matchListElement.innerHTML = "";
 
             matches.forEach((match: User) => {
                 matchListElement.appendChild(createMatchElement(match));
             })
+            } else {
+                //redirect to logged out homepage because user is not logged in
```

Comment:
nit: please update these comments to reflect the new changes

Reply:
Done!